Model/ApplePayOrder.php

```diff
@@ -0,0 +1,237 @@
+<?php
+/**
+ * Altapay Module for Magento 2.x.
+ *
+ * Copyright © 2018 Altapay. All rights reserved.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SDM\Altapay\Model;
+
+use SDM\Altapay\Api\OrderLoaderInterface;
+use Magento\Checkout\Model\Session;
+use Magento\CatalogInventory\Api\StockStateInterface;
+use Magento\CatalogInventory\Api\StockRegistryInterface;
+use SDM\Altapay\Api\TransactionRepositoryInterface;
+use SDM\Altapay\Model\SystemConfig;
+use Magento\Sales\Model\Order;
+use SDM\Altapay\Model\ConstantConfig;
+use SDM\Altapay\Model\Handler\CreatePaymentHandler;
+use Magento\Framework\DB\TransactionFactory;
+
+class ApplePayOrder {
+
+    /**
+     * @var OrderLoaderInterface
+     */
+    private $orderLoader;
+
+    /**
+     * @var Session
+     */
+    private $checkoutSession;
+
+    /**
+     * @var StockStateInterface
+     */
+    private $stockItem;
+
+    /**
+     * @var StockRegistryInterface
+     */
+    private $stockRegistry;
+
+    /**
+     * @var TransactionRepositoryInterface
+     */
+    private $transactionRepository;
+
+    /**
+     * @var SystemConfig
+     */
+    private $systemConfig;
+
+    /**
+     * @var Order
+     */
+    private $order;
+
+    /**
+     * @var CreatePaymentHandler
+     */
+    private $paymentHandler;
+
+    /**
+     * @var TransactionFactory
+     */
+    private $transactionFactory;
+
+    public function __construct(
+        OrderLoaderInterface $orderLoader,
+        Session $checkoutSession,
+        StockStateInterface $stockItem,
+        StockRegistryInterface $stockRegistry,
+        TransactionRepositoryInterface $transactionRepository,
+        SystemConfig $systemConfig,
+        Order $order,
+        CreatePaymentHandler $paymentHandler,
+        TransactionFactory $transactionFactory
+    ) {
+        $this->orderLoader           = $orderLoader;
+        $this->checkoutSession       = $checkoutSession;
+        $this->stockItem             = $stockItem;
+        $this->stockRegistry         = $stockRegistry;
+        $this->transactionRepository = $transactionRepository;
+        $this->systemConfig          = $systemConfig;
+        $this->order                 = $order;
+        $this->paymentHandler        = $paymentHandler;
+        $this->transactionFactory    = $transactionFactory;
+    }
+
+    /**
+     * @param                  $comment
+     * @param RequestInterface $request
+     *
+     * @throws AlreadyExistsException
+     */
+    public function handleCardWalletPayment($response, $order)
+    {
+        if ($response && $response->Result === 'Success' && isset($response->Transactions[0])) {
```

Comment:
You should never get a specific element of the transaction list by index id. You should assess what is the most updated transaction available in the payment, and use that one instead.